lib/mpoolmonitor/mpoolmonitor.go

```diff
@@ -76,7 +77,17 @@ func (mm *MpoolMonitor) startMonitoring(ctx context.Context) {
 // Otherwise, it inserts a new key value pair for the message. It removed any msgs not found in the
 // latest output of MpoolPending
 func (mm *MpoolMonitor) update(ctx context.Context) error {
-	t := time.Now()
+	localAddr := make(map[string]struct{})
+
+	addrs, err := mm.fullNode.WalletList(ctx)
+	if err != nil {
+		return fmt.Errorf("getting local addresses: %w", err)
+	}
+
+	for _, a := range addrs {
+		localAddr[a.String()] = struct{}{}
+	}
```

Comment:
I suggest we run this once on startup and that's it. No reason to call this API 3 times every epoch.

Reply:
Wallets are dynamic in Lotus and updating local wallets (dedicated wallets for miner actions) is not new. I would argue to reduce the frequency of this check but running it once on start up might not be the best solution. This would force folks to restart Boost nodes if they make any changes to daemon or miner in terms of wallets.

Reply:
Resolving based on https://filecoinproject.slack.com/archives/C037UKADRB2/p1688731687211039